lib/entities/parse_address_from_domain.dart

```diff
@@ -17,57 +20,60 @@ const unstoppableDomains = [
   'blockchain'
 ];
 
-Future<ParsedAddress> parseAddressFromDomain(
-    String domain, String ticker) async {
-  try {
-    final formattedName = OpenaliasRecord.formatDomainName(domain);
-    final domainParts = formattedName.split('.');
-    final name = domainParts.last;
-
-    if (domainParts.length <= 1 || domainParts.first.isEmpty || name.isEmpty) {
-      try {
-        final addresses = await fetchYatAddress(domain, ticker);
-
+Future<ParsedAddress> parse(String text, String ticker) async {
+    try {
+        if (text.hasOnlyEmojis) {
+        final yatService = getIt.get<YatService>();
```

Comment:
Move YatService to the class property and inject it through constructor

Reply:
done